docs(proxy) fix minor styling issues (#2366)
Kong/docs.konghq.com: app/2.1.x/proxy.md

Incoming requests with a matching hostname set in the TLS connection's SNI
extension would be routed to this Route. As mentioned, SNI routing applies not

> only to TLS, but also to other protocols carried over TLS - such as HTTPS and

Reviewer: ```suggestion
only to TLS, but also to other protocols carried over TLS (for example, HTTPS), and
```
